code/modules/vending/_vending.dm

```diff
@@ -1373,6 +1373,8 @@
 		LAZYREMOVE(item_record.returned_products, vended_item)
 		vended_item.forceMove(get_turf(src))
 	if(greyscale_colors)
+		if(greyscale_state)
+			vended_item.icon_state = greyscale_state
```

Comment:
This change is _extremely_ damaging to existing GAGS items' stability, making any item that shares a .json able to change between the other freely. Of note, this means jumpsuits to jumpskirts, without actually changing any data expected, only the icon state. This _also_ is going to cause issues with toggle-states, anything with a `_t` could be set as the base icon and cause errors.

I cannot endorse this change as it is. The benefit is far from worth the cost of maintenance this would entail. Please, find another solution; re-skins like the Detective's shotgun are the simplest that comes to mind

Reply:
ccing @ninjanomnom to give an actual verdict on this as they're the GAGS author and knows whether this is okay or not

Reply:
@OrionTheFox is completely correct and I would only add one thing: Messing with the icon state of an atom from code outside of that atom should in general be avoided if possible, though this is certainly not a hard rule. With generic systems like GAGS though you should be letting those objects handle the changes on their own if possible. You can see this from the rest of the code for GAGS in how it only touches variables that are specifically for GAGS and otherwise calls procs to update internal state.